# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
****

In [1]:
import pandas as pd
import numpy as np
from scipy import stats

In [41]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [42]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [43]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

## 3a. What test is appropriate for this problem? 
## 3b. Does CLT apply?

In [44]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4870 entries, 0 to 4869
Data columns (total 65 columns):
id                    4870 non-null object
ad                    4870 non-null object
education             4870 non-null int8
ofjobs                4870 non-null int8
yearsexp              4870 non-null int8
honors                4870 non-null int8
volunteer             4870 non-null int8
military              4870 non-null int8
empholes              4870 non-null int8
occupspecific         4870 non-null int16
occupbroad            4870 non-null int8
workinschool          4870 non-null int8
email                 4870 non-null int8
computerskills        4870 non-null int8
specialskills         4870 non-null int8
firstname             4870 non-null object
sex                   4870 non-null object
race                  4870 non-null object
h                     4870 non-null float32
l                     4870 non-null float32
call                  4870 non-null float32
city        

In [45]:
df = data[['race', 'call']]

In [46]:
df.describe()

call
count  4870.000000
mean      0.080493
std       0.272079
min       0.000000
25%       0.000000
50%       0.000000
75%       0.000000
max       1.000000

In [47]:
df_b = df[df['race'] == 'b']
df_w = df[df['race'] == 'w']

In [48]:
mean_b_call = df_b['call'].sum() / len(df_b)
print(mean_b_call)

0.064476386037


In [49]:
mean_w_call = df_w['call'].sum() / len(df_w)
print(mean_w_call)

0.0965092402464


We don't know the population standard deviation and so we will use the t statistic.  Yes, CLT applies.

## 2. What are the null and alternate hypotheses?

Null Hypothesis: Black sounding names and White sounding names have the same callback rate

Alternate Hypothesis: Black sounding names and White sounding names have different callback rates

## 3. Compute margin of error, confidence interval, and p-value.

In [50]:
conf_int = np.percentile(df['call'], [2.5, 97.5])
conf_int

array([ 0.,  1.])

In [61]:
se = np.sqrt((mean_b_call*(1-mean_b_call)/len(df_b))+(mean_w_call*(1-mean_w_call)/len(df_w)))
margin_of_error = (1.962*se)
print("margin of error: ", round(margin_of_error, 5))

margin of error:  0.01527


In [59]:
b_clo = mean_b_call - margin_of_error
b_chi = mean_b_call + margin_of_error

print("Black-sounding names confidence interval: ", round(b_clo, 4), ", ", round(b_chi, 4))

w_clo = mean_w_call - margin_of_error
w_chi = mean_w_call + margin_of_error

print("White-sounding names confidence interval: ", round(w_clo, 4), ", ", round(w_chi, 4))


Black-sounding names confidence interval:  0.0492 ,  0.0797
White-sounding names confidence interval:  0.0812 ,  0.1118


In [69]:
import scipy
from scipy import stats

x, y = scipy.stats.ttest_ind(df_b['call'], df_w['call'])
print("p-value: ", y)

p-value:  3.94080210313e-05


## 4. Write a story describing the statistical significance in the context of the original problem.

Black sounding names had a callback rate of 6.4% while White sounding names had a callback rate of 9.7%.  The 3.3% difference along with the low p-value of 0.000039 suggest the null hypothesis is wrong and that race does indeed have an impact on the US job market.  

## 5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

The analysis above suggests race is a factor in callback success but does not prove it's the "most important factor."  In order to identify significant factors in callback success, one would have to investigate and analyze other variables in the data to include education, years of experience, skills, etc.  